runtime/desired.go

```diff
@@ -0,0 +1,198 @@
+package runtime
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	xfnv1alpha1 "github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DesiredResources struct {
+	resources []Resource
+	composite xfnv1alpha1.DesiredComposite
+}
+
+// GetFromKubeObject gets the k8s resource o from a provider kubernetes object kon (Kube Object Name)
+// from the desired array of the FunctionIO.
+func (d *DesiredResources) GetFromKubeObject(ctx context.Context, o client.Object, kon string) error {
```

Comment:
Nitpick: The methods signatures are not consistent. `GetFromKubeObject(ctx, object, name)` and `GetManagedResource(ctx, name, object)` (the order or name and object)

Reply:
Good catch